paths_get.py

```diff
@@ -0,0 +1,168 @@
+from typing import Iterable, Tuple, Union, Literal, Sequence, Dict
+
+import numpy as np
+import pandas as pd
+from anndata import AnnData
+from anndata._core.raw import Raw
+
+
+# TODO: allow sequences: ("obsm", "X_pca", [0, 1])
+AnnDataPath = Union[
```

Comment:
Could this just be a real class? Possibly an `ElementRef`?

Reply:
Sure! Before making it a class, we need to think about how normalizing works with or without an AnnData object. For now I have no fallback behavior, everything is deterministic and independent of AnnData content.

For a practical usage we’d need at least some limited fallback and therefore binding to an AnnData object.

Reply:
I think it would be fair to have an abstract reference and an associated reference. Some things you don't need the anndata to know, like which attribute, and whether it's valid. In non-`uns` cases (the aligned mappings and X) you can also tell what the indexing statement is.

Reply:
OK, for later, we just have the abstract one now